presets/presetManager.py

```diff
@@ -0,0 +1,272 @@
+import os.path
+import logging
+import json
+
+logger = logging.getLogger(__name__)
+
+
+class PresetManager(object):
+    '''PresetManager deals with presets loading, validating, storing
+
+    you can use it like this:
+
+    python::
+    pm = PresetManager(["/path/to/presets/folder", "/another/path"])
+    '''
+
+    MAX_DEPTH = 2
+
+    def __init__(self, paths, strict=False):
+        """load, validate, and store presets under ```self.presets```
+
+        :param paths:  list of paths to preset file
+                       or folder containing presets.
+        :param strict: if strict is False
+                       all exceptions and errors
+                       will be converted to log messages.
+        """
+
+        self.presets = {}
+        self.strict = strict
+
+        # normalize paramter type string -> array
+        if isinstance(paths, basestring):
+            paths = [paths]
+
+        self._loadPaths(paths)
+
+    def _loadPaths(self, paths, depth=0):
+        '''
+        Goes recursevly through the given list of paths
+        in order to find and pass all preset files to ```_loadPreset()```
+        '''
+
+        if depth > self.MAX_DEPTH:
+            return
+
+        for p in paths:
+            path = p
```

Comment:
@boyska you really need to be more specific!

You must keep in mind that I wrote this code, it makes sense for me and I did also tests.

If you find something strange, you have to explain me which part and why it is strange for you. 

> Can you explain why are you doing something like that? I guess there's a reason, and I cannot understand it.

In the previous comment there is not explicit subject and it's very difficult for me to guess what are you talking about. I've tried but I really can't.